docs/data/toolpad/core/components/app-provider/AppProviderTheme.tsx

```diff
@@ -0,0 +1,177 @@
+import * as React from 'react';
+import Box from '@mui/material/Box';
+import Typography from '@mui/material/Typography';
+import { createTheme, Theme } from '@mui/material/styles';
+import DashboardIcon from '@mui/icons-material/Dashboard';
+import TimelineIcon from '@mui/icons-material/Timeline';
+import { AppProvider } from '@toolpad/core/AppProvider';
+import { DashboardLayout } from '@toolpad/core/DashboardLayout';
+import type { Navigation, Router } from '@toolpad/core';
+
+const NAVIGATION: Navigation = [
+  {
+    kind: 'header',
+    title: 'Main items',
+  },
+  {
+    slug: '/page',
+    title: 'Page',
+    icon: <DashboardIcon />,
+  },
+  {
+    slug: '/page-2',
+    title: 'Page 2',
+    icon: <TimelineIcon />,
+  },
+];
+
+const defaultDarkTheme = createTheme({ palette: { mode: 'dark' } });
+
+const customTheme: Theme = createTheme(defaultDarkTheme, {
```

Comment:
I really love the look of this demo, but I feel like the MUI theme is the wrong vehicle to configure these customizations. These are customizations that apply to the application as a whole. I feel like we should look more into a `slots`/`slotProps` setup similar to how X [promotes it](https://mui.com/x/react-date-pickers/custom-field/). That way you can target custumizations e.g. the List under the dashboard layout only.

Reply:
Didn't fully understand this comment, this file here is for exemplifying how to set up a custom theme for the whole application, mostly some palette colors... Do you mean this just about the `components` in this theme example?